sapi/cli/php_cli.c

```diff
@@ -1329,7 +1350,15 @@ int main(int argc, char *argv[])
 
 	if (sapi_module == &cli_sapi_module) {
 		if (ini_entries) {
-			ini_entries = realloc(ini_entries, ini_entries_len + sizeof(HARDCODED_INI));
+			char *pini_entries;
+			if ((ini_entries_len + sizeof(HARDCODED_INI)) >= SIZE_MAX) {
+				goto out;
+			}
+			pini_entries = realloc(ini_entries, ini_entries_len + sizeof(HARDCODED_INI));
+			if (!pini_entries) {
+				goto out;
+			}
+			ini_entries = pini_entries;
```

Comment:
Using the suggestion from https://github.com/php/php-src/pull/2162#issuecomment-255072565, this whole change (including the overflow check) could be written as:

```
ini_entries = safe_perealloc(ini_entries, ini_entries_len, 1, sizeof(HARDCODED_INI));
```